### PR TITLE
ifp: handle placement blockages

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/ifp/src/InitFloorplan.tcl [hunk 1]
```diff
@@ -263,4 +269,33 @@ proc microns_to_mfg_grid { microns } {
...
+
+  # Gather rows needing to be cut because of placement blockages
+  set rows_to_cut []
+  set row_placement_blockages [dict create]
+  foreach blockage $placement_blockages {
+    if {![$blockage isSoft]} {
+      foreach row [$block getRows] {
+        if {[tap::overlaps $blockage $row 0 0]} {
+          set row_name [$row getName]
+          if {![dict exists $row_placement_blockages $row_name]} {
```
Comment (QuantamHD) [inline src/ifp/src/InitFloorplan.tcl:283]: This check should move before the overlap check. If the row is already removed there's no benefit to checking if it overlaps.

File: src/ifp/src/InitFloorplan.tcl [hunk 2]
```diff
@@ -263,4 +269,33 @@ proc microns_to_mfg_grid { microns } {
   }  
 }
 
+proc cut_rows {block placement_blockages} {
```
Comment (QuantamHD) [inline src/ifp/src/InitFloorplan.tcl:272]: Do you think there would be a benefit to move this to C++ so we could reuse some of the unit testing infrastructure there? It seems in general the project is moving away from TCL for top level logic.

File: src/ifp/src/InitFloorplan.tcl (hunk 1 above)
Comment (QuantamHD) [inline src/ifp/src/InitFloorplan.tcl:283]: This check should move before the overlap check, and if it's already removed the loop should continue. If the row is already removed there's no benefit to checking if it overlaps.

File: src/ifp/src/InitFloorplan.tcl (hunk 2 above)
Comment (jjcherry56) [reply src/ifp/src/InitFloorplan.tcl:272]: The unit testing is language agnostic.

File: src/ifp/src/InitFloorplan.tcl [hunk 3]
```diff
@@ -263,4 +269,33 @@ proc microns_to_mfg_grid { microns } {
...
+    }
+  }
+
+  # cut rows around placement blockages
+  foreach row $rows_to_cut {
+    tap::cut_row $block $row $row_placement_blockages 0 0 0
+  }
+
+  utl::info "IFP" 21 "Cut rows: [llength $rows_to_cut]"
+  utl::info "IFP" 22 "Rows after cut: [llength [$block getRows]]"
```
Comment (jjcherry56) [inline src/ifp/src/InitFloorplan.tcl:298]: what is the point of printing rows after cut?
ifp already says how many rows there are

**Review by jjcherry56 — APPROVED** [review]

**Pull request merged**